test/backing_queue_SUITE.erl

```diff
@@ -0,0 +1,1689 @@
+%% The contents of this file are subject to the Mozilla Public License
+%% Version 1.1 (the "License"); you may not use this file except in
+%% compliance with the License. You may obtain a copy of the License at
+%% http://www.mozilla.org/MPL/
+%%
+%% Software distributed under the License is distributed on an "AS IS"
+%% basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+%% License for the specific language governing rights and limitations
+%% under the License.
+%%
+%% The Original Code is RabbitMQ.
+%%
+%% The Initial Developer of the Original Code is GoPivotal, Inc.
+%% Copyright (c) 2011-2017 Pivotal Software, Inc.  All rights reserved.
+%%
+
+-module(backing_queue_SUITE).
+
+-include_lib("common_test/include/ct.hrl").
+-include_lib("kernel/include/file.hrl").
+-include_lib("amqp_client/include/amqp_client.hrl").
+
+-compile(export_all).
+
+-define(PERSISTENT_MSG_STORE, msg_store_persistent).
+-define(TRANSIENT_MSG_STORE,  msg_store_transient).
+
+-define(TIMEOUT, 30000).
+
+-define(VARIABLE_QUEUE_TESTCASES, [
+    variable_queue_dynamic_duration_change,
+    variable_queue_partial_segments_delta_thing,
+    variable_queue_all_the_bits_not_covered_elsewhere_A,
+    variable_queue_all_the_bits_not_covered_elsewhere_B,
+    variable_queue_drop,
+    variable_queue_fold_msg_on_disk,
+    variable_queue_dropfetchwhile,
+    variable_queue_dropwhile_varying_ram_duration,
+    variable_queue_fetchwhile_varying_ram_duration,
+    variable_queue_ack_limiting,
+    variable_queue_purge,
+    variable_queue_requeue,
+    variable_queue_requeue_ram_beta,
+    variable_queue_fold,
+    variable_queue_batch_publish,
+    variable_queue_batch_publish_delivered
+  ]).
+
+-define(BACKING_QUEUE_TESTCASES, [
+    bq_queue_index,
+    bq_queue_index_props,
+    bq_variable_queue_delete_msg_store_files_callback,
+    bq_queue_recover
+  ]).
+
+-define(BACKING_QUEUE_PARALLEL, [
+    {variable_queue_default, [parallel], ?VARIABLE_QUEUE_TESTCASES},
+    {variable_queue_lazy, [parallel], ?VARIABLE_QUEUE_TESTCASES ++
+                              [variable_queue_mode_change]}
+  ]).
+
+all() ->
+    [
+      {group, backing_queue_tests}
+    ].
+
+groups() ->
+    [
+     {backing_queue_tests, [],
+      [
+       {backing_queue_embed_limit_0, [], ?BACKING_QUEUE_PARALLEL},
+       {backing_queue_embed_limit_1024, [], ?BACKING_QUEUE_PARALLEL}
+      ]},
+     {backing_queue_tests, [], [
+          msg_store,
+          {backing_queue_embed_limit_0, [], ?BACKING_QUEUE_TESTCASES},
+          {backing_queue_embed_limit_1024, [], ?BACKING_QUEUE_TESTCASES}
+        ]}
+    ].
```

Comment:
Is there a reason why you splitted that into two groups with the same name? Is it to get a dedicated broker/cluster for each top-level group?